src/extension.ts

```diff
@@ -1068,9 +1066,6 @@ function launchMetals(
       scalaDebugger.initialize(outputChannel).forEach((disposable) => {
         context.subscriptions.push(disposable);
       });
-      client.onNotification(DecorationTypeDidChange.type, (options) => {
-        decorationType = window.createTextEditorDecorationType(options);
-      });
       client.onNotification(DecorationsRangesDidChange.type, (params) => {
         const editors = window.visibleTextEditors;
         const path = Uri.parse(params.uri).toString();
```

Comment:
I've noticed that `client.onNotification` return `Disposable` but we're discarding returned value. Maybe it should be added to `context.subscriptions` (yeah, it's a broader problem)

`extension.ts` is a very large file, I wonder if we can e.g. create `notifications` dir and extract all `client.onNotification` to different files

Reply:
I added all the disposables from `onNotification` to subscriptions. Good idea!

Reply:
Thanks, that's a step into good direction ;)